simplecpp.cpp

```diff
@@ -1938,11 +1962,15 @@ static bool realFileName(const std::string &f, std::string *result)
     return true;
 }
 
+static ThreadSafeCache<std::string, std::string> realPathMap;
```

Comment:
It seems strange to me that we want to have different maps for different `realFileName` functions. Do you see wrong behavior when reusing the same map?